### ISSUE TITLE
Form and fields "hasError" after clear and/or reset

### ISSUE THREAD

**Issue by pdfowler** [foxhound87/mobx-ajv-form#289]
Is this the expected behavior? I would expect the form to go back to its base state after calling clear or reset, but the form and the required fields re-validate after submit => clear.

Form state after calling clear + reset after a successful form submit:
```
submitting:false
validating:false
hasError:true
isValid:false
isDirty:false
isPristine:true
isDefault:true
isEmpty:true
touched:false
changed:false
```

**Comment by foxhound87** [comment]
The `submit` action will validate the fields, `clear`/`reset` actions doesn't revalidate them.
The `hasError` prop is true and `isValid` prop is false if the field `error` is defined.
In your case, the issue seems related to #273

**Comment by mdansei** [comment]
Same thing here, the reason being I think is when clearing or resetting the form this will trigger onChange handler, this happens when you have validateOnChange set to true. As a workaround for this that worked for me : 
```
form.state.options.set({validateOnChange: false})
form.clear() // or form.reset()
form.state.options.set({validateOnChange: true})
```

**Comment by foxhound87** [comment]
The fields are now validated also on `reset` and on `clear` and  `isValid`/`error` are always synced with the validation rules. This was required to fix #273

**Comment by ethanyou725** [comment]
Now, it seems link `clear()` or `reset()` trigger `validateOnBlur`, my workaround:
```
 class MyForm extends Form {
  plugins() {
    return { vjf: validator };
  }
  options() {
    return {
      showErrorsOnReset: false,
      validateOnChange: true,
      validateOnBlur: false
    };
  }
  setup() {
    return {
      fields: initFields
    };
  }
}

```
